tox.ini

```diff
@@ -11,10 +11,17 @@ setenv = PYDEVD_DISABLE_FILE_VALIDATION=1
 commands = test-docs-notebooks {posargs} --config-path scripts/config/notebook-testing.toml
 
 [testenv:{lint,fix}]
-deps = squeaky==0.7.0
-commands = 
-  lint: squeaky --check --no-advice {posargs:docs}
-  fix:  squeaky {posargs:docs}
+deps = 
+  squeaky==0.7.0
+  ruff==0.7.1
+
+commands =
+    lint: ruff check {posargs:docs}
+    lint: ruff format --check {posargs:docs}
+    lint: squeaky --check --no-advice {posargs:docs}
+    fix: squeaky {posargs:docs}
+    fix: ruff format {posargs:docs}
+    fix: ruff check --fix {posargs:docs}
```

Comment:
Unlike `ruff format` and `squeaky`, `ruff check` can't always be autofixed. This affects the order we want to call these commands.

For `lint` (i.e. check but don't fix), we want to show anything requiring manual attention first. Formatting can be done automatically. This is the other way round for `fix`: We want to fix everything we can, then hand over to the user for anything requiring manual attention.